trillian.proto

```diff
@@ -194,20 +194,20 @@ message SignedLogRoot {
   int64 tree_revision = 6;
 }
 
-message MapperMetadata {
-  bytes source_log_id = 1;
-  int64 highest_fully_completed_seq = 2;
-  int64 highest_partially_completed_seq = 3;
- }
-
 // SignedMapRoot represents a commitment by a Map to a particular tree.
 message SignedMapRoot {
   int64 timestamp_nanos = 1;
   bytes root_hash = 2;
-  MapperMetadata metadata = 3;
+  reserved 3;  // was MapperMetadata (removed, replaced by 'Any metadata').
 	// TODO(al): define serialized format for the signature scheme.
   sigpb.DigitallySigned signature = 4;
-
   int64 map_id = 5;
   int64 map_revision = 6;
+  // Metadata associated with the Map root.  This is set by the map personality
+  // layer in a SetMapLeavesRequest request.  It could be used to persist state
+  // needed to recreate the Map from an external data source. 'Any' can hold
```

Comment:
That's good, thanks. I think it's fine to drop the "how any works section", you've answered the important questions by saying how data gets here (SetMapLeaves) and who's responsible for it (map personality).

A final q: is it OK to not have any metadata? (I assume so, but you may want to mention whether it's optional.)

Reply:
Do you think the bit about "the Map personality should know what types it has previously assigned in SetMapLeavesRequest calls" is implied and could be omitted too?

The 'ok to not have metadata' question is something I want to leave for the next PR.  That's going to depend on how (or if) we decide to represent the use of map metadata in the Admin CreateTree API.

Reply:
1. Yes, I think it's fine to imply that.

2. Fair enough.